services/admin/src/store.ts

```diff
@@ -277,16 +285,55 @@ export class Store {
 
         const lines = readline.createInterface(fs.createReadStream(filePath));
 
+        const currentCvrFileMode =
+          this.getCurrentCvrFileModeForElection(electionId);
+        let lastLineCvrMode: Optional<Admin.CvrFileMode>;
+        let nextLineCvrMode: Optional<Admin.CvrFileMode>;
+
         for await (const line of lines) {
           if (line.trim() === '') {
             continue;
           }
 
-          // TODO(https://github.com/votingworks/vxsuite/issues/2716): add error checking against
-          // attempts to mix "live" and "test" CVRs for the same election.
+          // Parse CVR data for validation.
+          const cvr = safeParseJson(line).unsafeUnwrap() as CastVoteRecord;
+          nextLineCvrMode = cvr._testBallot
+            ? Admin.CvrFileMode.Test
+            : Admin.CvrFileMode.Official;
+
+          // Ensure CVR mode matches previous file imports, if any:
```

Comment:
Do we need to check this for every line? Maybe we could just check this once at the end? (Again, probably not a necessary optimization, but it's in the critical path)

Reply:
true, this feels like it could just go at the end